messages/src/utils/error.rs

```diff
@@ -1,621 +1,69 @@
-        insert_c_message(&mut m, &INVALID_STATE);
-        insert_c_message(&mut m, &DID_ALREADY_EXISTS_IN_WALLET);
-        insert_c_message(&mut m, &DUPLICATE_MASTER_SECRET);
-        insert_c_message(&mut m, &INVALID_LEDGER_RESPONSE);
-        insert_c_message(&mut m, &THREAD_ERROR);
-        insert_c_message(&mut m, &INVALID_PROOF_REQUEST);
-        insert_c_message(&mut m, &INVALID_REVOCATION_DETAILS);
-        insert_c_message(&mut m, &INVALID_REV_REG_DEF_CREATION);
-        insert_c_message(&mut m, &INVALID_REVOCATION_TIMESTAMP);
-        insert_c_message(&mut m, &INVALID_REV_ENTRY);
-        insert_c_message(&mut m, &DUPLICATE_SCHEMA);
-        insert_c_message(&mut m, &UNKNOWN_SCHEMA_REJECTION);
-        insert_c_message(&mut m, &UKNOWN_LIBINDY_TRANSACTION_REJECTION);
-        insert_c_message(&mut m, &MISSING_PAYMENT_METHOD);
-        insert_c_message(&mut m, &LOGGING_ERROR);
-        insert_c_message(&mut m, &INVALID_ATTACHMENT_ENCODING);
-        insert_c_message(&mut m, &UNKNOWN_ATTACHMENT_ENCODING);
-        insert_c_message(&mut m, &UNKNOWN_MIME_TYPE);
-        insert_c_message(&mut m, &ACTION_NOT_SUPPORTED);
-        insert_c_message(&mut m, &INVALID_REDIRECT_DETAILS);
-        insert_c_message(&mut m, &NO_AGENT_INFO);
-        insert_c_message(&mut m, &REV_REG_DEF_NOT_FOUND);
-        insert_c_message(&mut m, &REV_DELTA_NOT_FOUND);
-        insert_c_message(&mut m, &POISONED_LOCK);
-        insert_c_message(&mut m, &CREATE_PUBLIC_AGENT);
-        insert_c_message(&mut m, &INVALID_MESSAGE_FORMAT);
-        insert_c_message(&mut m, &CREATE_OUT_OF_BAND);
-        insert_c_message(&mut m, &CREATE_AGENT);
-        insert_c_message(&mut m, &REV_DELTA_FAILED_TO_CLEAR);
-        m
-    };
+pub mod prelude {
+    pub use super::{err_msg, MessagesError, MesssagesErrorKind, MessagesResult};
 }
 
-// ******* END *******
-
-// Helper function for static defining of error messages. Does limited checking that it can.
-fn insert_c_message(map: &mut HashMap<u32, CString>, error: &Error) {
-    if map.contains_key(&error.code_num) {
-        panic!("Error Code number was repeated which is not allowed! (likely a copy/paste error)")
-    }
-    map.insert(error.code_num, CString::new(error.message).unwrap());
+#[derive(Copy, Clone, Eq, PartialEq, Debug, thiserror::Error)]
+pub enum MesssagesErrorKind {
+    #[error("Object is in invalid state for requested operation")]
+    InvalidState,
+    #[error("Invalid JSON string")]
+    InvalidJson,
+    #[error("IO Error, possibly creating a backup wallet")]
+    IOError,
+    #[error("Unable to serialize")]
+    SerializationError,
+    #[error("Invalid DID")]
+    InvalidDid,
+    #[error("Invalid verkey")]
+    InvalidVerkey,
+    #[error("Invalid URL")]
+    InvalidUrl,
+    #[error("Value needs to be base58")]
+    NotBase58,
 }
 
-#[derive(Clone, Copy)]
-pub struct Error {
-    pub code_num: u32,
-    pub message: &'static str,
+#[derive(Debug, thiserror::Error)]
+pub struct MessagesError {
```

Comment:
nitpick: but should this (and most contents of this file) be within a `src/error.rs` file instead - just for consistency across the crates? i.e. like `aries_vcx` and `agency_client`. as this isn't really a `utils` IMO. but let me know what you think

Reply:
Approach unified in error refactoring PR https://github.com/hyperledger/aries-vcx/pull/702